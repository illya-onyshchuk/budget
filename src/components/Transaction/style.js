import styled from 'styled-components'

export const Wrapper = styled.div`
  background-color:  ${({value}) => value < 0 ? '#cf9e9e' : '#3333'};
  border: 1px solid #3a2323;
  border-radius: 10px;
  padding: 5px;
  margin-bottom: 10px;
`

Wrapper.displayName = 'TransactionWrapper'