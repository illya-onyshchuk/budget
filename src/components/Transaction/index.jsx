import PropTypes from 'prop-types'
import { Wrapper, TransactiononDate, Value, Comment } from './style';

const Transaction = ({ transaction: {value, date, comment } }) => {
  return (
    <Wrapper value={value}>
      <TransactiononDate>{date}</TransactiononDate>
      <Value>{value.toFixed(3)}</Value>
      <Comment>{comment}</Comment>
    </Wrapper>
  );
}

Transaction.prototype = {
  transaction: PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.number,
  })
};

Transaction.defaultProps = {
  transaction: {
    label: '',
    value: 0,
  }
};

export default Transaction;

