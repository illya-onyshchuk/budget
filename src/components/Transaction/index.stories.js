
import Transaction from ".";

export default {
  title: 'Example/Transaction',
  component: Transaction,
};

const Template = (args) => <Transaction {...args} />

export const Income =  Template.bind({});
Income.args = {
  transaction: {
    label: 'some income',
    value: 23, 
  },
}
