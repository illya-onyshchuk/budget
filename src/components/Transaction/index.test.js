import React  from "react";
import Transaction from '.';
import { shallow } from 'enzyme';

describe('Transaction component', () => {
  let sut;
  let props;


  beforeEach(() => {
    props = {
      transaction: {
        value: 23,
        date: '24.04.2023',
        comment: 'Some coments' 
      }
    }

    sut = shallow(<Transaction {...props}/>);
  });

  it('should show transaction', () => {
      expect(sut).toMatchSnapshot();
    });

  it('should show 2 zeros after amount', () => {
      expect(sut.find('Value').text()).toBe('23.00')
  });
}) 