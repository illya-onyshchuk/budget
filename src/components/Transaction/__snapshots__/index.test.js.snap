// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `
<TransactionWrapper
  value={23}
>
  <TransactiononDate>
    24.04.2023
  </TransactiononDate>
  <Value>
    23.00
  </Value>
  <Comment>
    Some coments
  </Comment>
</TransactionWrapper>
`;
